provider/import.go

```diff
@@ -130,6 +130,16 @@ func Import(ctx context.Context, p Provider, hcl, tfstate writer.Writer, f *filt
 						return errors.Wrapf(err, "error while calculating the satate of resource %q", t)
 					}
 				}
+				// we construct a map[string]string to perform
+				// interpolation later. By default, the value to interpolate is the ID
+				// but in some case (ex: google), it can be different.
+				state := r.InstanceState()
+				if state != nil {
+					key := getKey(state)
+					value := state.Attributes[key]
+					interpolatedValue := fmt.Sprintf("${%s.%s.%s}", r.Type(), r.Name(), key)
+					interpolation[value] = interpolatedValue
+				}
```

Comment:
~Also all this logic could be done on the `r.HCL` for example as it has all the logic and the actual scope for all this information (state and Name and all of them)~

Maybe better have a method on the `Resource` that actually returns all the possible interpolated values (right now just ID but potentially more) as the Resource has access to the schema and could potentially know them.

But maybe for now it's ok this way.